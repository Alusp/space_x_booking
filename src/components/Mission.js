import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { displayMission, joinmission, leavemission } from '../redux/mission/mission';
import './mission.css';

const Mission = () => {
  const allmission = useSelector((state) => state.mission);
  const dispatch = useDispatch();

  const clickHandler = (id, join) => {
    if (join === false || join === undefined) {
      dispatch(joinmission(id));
    } else {
      dispatch(leavemission(id));
    }
  };
  useEffect(() => {
    if (!allmission.length) dispatch(displayMission());
  }, [allmission.length, dispatch]);

  return (
    <div className="table-container">
      <table>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </tr>
        {allmission.map((mission) => (
          <tr key={mission.mission_id}>
            <td>{mission.mission_name}</td>
            <td>{mission.description}</td>
            <td className="button-container">
              <button type="button" className={mission.join ? 'memberac' : 'member'}>{mission.join ? 'ACTIVE MEMBER' : 'NOT A MEMBER '}</button>
            </td>

            <td className="button-container1">
              <button id={mission.mission_id} onClick={() => clickHandler(mission.mission_id, mission.join)} type="button" className={mission.join ? 'joinac' : 'join'}>{mission.join ? 'Leave Mission' : 'Join Mission'}</button>
            </td>

          </tr>
        ))}
      </table>
    </div>
  );
};

export default Mission;
