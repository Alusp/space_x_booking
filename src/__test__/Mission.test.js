import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Mission />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
