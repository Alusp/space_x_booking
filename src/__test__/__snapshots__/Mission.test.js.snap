// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="table-container"
>
  <table>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </table>
</div>
`;
